File: kotogram/reversing_codec.py
"""Implementation of a reversing codec."""

from .codec import Codec


class ReversingCodec(Codec):
    """A codec that reverses strings for both encoding and decoding."""

    def encode(self, text: str) -> str:
        """Encode a string by reversing it.

        Args:
            text: The string to encode

        Returns:
            The reversed string
        """
        return text[::-1]

    def decode(self, text: str) -> str:
        """Decode a string by reversing it.

        Args:
            text: The string to decode

        Returns:
            The reversed string
        """
        return text[::-1]

File: src/reversing-codec.ts
import { Codec } from "./codec.js";

/**
 * A codec that reverses strings for both encoding and decoding.
 */
export class ReversingCodec implements Codec {
  /**
   * Encode a string by reversing it.
   * @param text - The string to encode
   * @returns The reversed string
   */
  encode(text: string): string {
    return text.split("").reverse().join("");
  }

  /**
   * Decode a string by reversing it.
   * @param text - The string to decode
   * @returns The reversed string
   */
  decode(text: string): string {
    return text.split("").reverse().join("");
  }
}
